topology/probes/istio/istio.go

```diff
@@ -58,6 +56,13 @@ func NewIstioProbe(g *graph.Graph) (*k8s.Probe, error) {
 
 	k8s.InitSubprobes(enabledSubprobes, subprobeHandlers, client, g, Manager)
 
+	verifierHandlers := []verifierHandler{
+		newVirtualServiceVerifier,
+		newGatewayVerifier,
```

Comment:
I was expecting just one verifier:

```
newVirtualServiceGatewayVerifier
```

which is "smart" enough to track both virtualservice events as well as gateway events.

Reply:
Your suggestion requires one of the following inefficient implementations: 
Either to create many unnecessary `resourceVerifiers` - one for each couple of resources (as happens in the links case), or, alternatively, to listen to multiple resources events by the same `resourceVerifier`. In my opinion, both of the implementations are wasteful, since they don't take advantage of the fact that a verification procedure can be done only using one resource (as opposed to the links case).

Reply:
DONE